### PR TITLE
fix(project): image link, update CSS, add code highlighter

### PR DESCRIPTION
**Pull request by lasjorg**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.

- [x] Closes currently open issue: Closes https://github.com/freeCodeCamp/freeCodeCamp/issues/42226

#### Description

So I was going to just update the broken image link but then I noticed some issues with the responsiveness and started to change a few things. Before you know it it's a bit of a "refresh". I have updated the CSS and added a code highlighter as well just for good measure. The code highlighter implementation is likely not super well optimized (runs on componentDidUpdate) but I think it's fine.

Here is a Codepen fork for preview as well
https://codepen.io/lasjorg/pen/jOBGzgP

---

@scissorsneedfoodtoo When you update the Codepen (I assume it will be you) can you also set the HTML Preprocessor to none (click the cog on the left of the HTML box > set HTML Preprocessor from Haml to none). Not sure why it's there but it shouldn't be needed.

There is also a CSS file linked to in the Codepen Settings (Settings > CSS) which I don't believe is being used for anything.

`https://cdnjs.cloudflare.com/ajax/libs/mocha/3.0.2/mocha.min.css`

It looks like some styles for mocha but the file isn't in any of the other projects and I can't see where its style would be used. It might just be some old CSS for an older version of the test script, not sure.

### CODE REVIEW

**Comment by scissorsneedfoodtoo** [comment]
Hi @lasjorg, thanks for your hard work on this and for taking the time to write such helpful instructions.

I took a look at this and noticed a couple of things, one which may need some discussion. I'll leave them as review comments.

File: src/projects/markdown-previewer/index.jsx
Comment (scissorsneedfoodtoo) [inline src/projects/markdown-previewer/index.jsx:24]: Very minor nitpick, but this could probably be simplified to:

```suggestion
  return `<pre><code class="language-javascript">${code}</code></pre>\n`;
```

File: src/projects/markdown-previewer/index.jsx
Comment (scissorsneedfoodtoo) [inline src/projects/markdown-previewer/index.jsx:20]: If you agree with the change below, would you mind also simplifying this line a bit?:

```suggestion
  return `<a target="_blank" href="${href}">${text}</a>`;
```

**Comment by scissorsneedfoodtoo** [comment]
Also, since you're making a lot of improvements to this project, let's also update the link on line 141 to point to https://www.freecodecamp.org/.

**Comment by scissorsneedfoodtoo** [comment]
The last thing I noticed is a weird bug where Prism seems to add some extra whitespace to the `class` attribute, which throws off one of the tests in certain situations.

To reproduce this, expand either the editor or the preview, then run the test. Then test 6 which reads "When my markdown previewer first loads, the default markdown in the #editor field should be rendered as HTML in the #preview element" will fail.

This seems to be because Prism adds an extra space after rendering the code highlights.

Our test suite gets the `innerHTML` of the preview pane before the tests are run, then compares it again with the preview pane's `innerHTML` when it gets to test 6.

Sometime between those two points, if either the preview or editor pane are maximized, Prism renders `class=\"  language-javascript\">` with two spaces, not `class=\" language-javascript\">` with a single space like the test expects. It would be better if it didn't render extra spaces at all, but it seems to be a [known issue](https://github.com/PrismJS/prism/issues/2383).

It's a pretty minor thing, and I'm not sure how many people will actually run into this. And if you press the "Run Tests" button again, everything passes, at least until the editor or preview pane are resized again.

It might be possible to strip off the extra whitespace introduced by Prism in test itself, but I can see that possibly introducing some edge cases.

Then again, I'm thinking we might just leave this as-is since resizing the panes isn't a requirement. I don't think learners will run into this in their own projects if they don't add that feature.

What do think @lasjorg? Is there a simple way to prevent Prism from adding that extra whitespace? Or should we just leave this the way it is for now?

**Comment by lasjorg** [comment]
@scissorsneedfoodtoo Totally agree with everything. I was paying too much attention to the CSS, and at the last minute just threw in the code highlighter. Your thorough review is much appreciated.

I think I came up with a solution that fixes the class name issue.

I also just realized that the version of React and Marked we are using are pretty old. I updated them and everything still seems to work just fine. But I can revert them back if needed.

---


Here is an updated Codepen as well
https://codepen.io/lasjorg/pen/VwpygEo

**Comment by scissorsneedfoodtoo** [comment]
@lasjorg, no worries. I like the addition of the code highlights, even if it's not a requirement. If anything it's another cool feature that learners can choose to implement themselves if they want.

Also, I checked your updated pen and locally and no longer run into that issue where resizing the editor or preview pane causes the tests to fail. Thanks for diving back in and fixing that.

Everything LGTM :+1: :+1:. Will just wait for another review before merging this and deploying your changes to CodePen with your helpful instructions above.

**Comment by moT01** [comment]
Hey, sorry I missed this one - I'll try and give it a look in the next day or two.

**Review by moT01 — APPROVED** [review]
LGTM @lasjorg 🎉 Thanks for fixing this up - it looks a lot better 👍 

Want to take a quick look at [the pen](https://codepen.io/freeCodeCamp/pen/GrZVVO) and make sure it's good?

**Comment by lasjorg** [comment]
> Want to take a quick look at [the pen](https://codepen.io/freeCodeCamp/pen/GrZVVO) and make sure it's good?

@moT01 Looks good.

**Pull request merged**
